xskillscore/core/contingency.py

```diff
@@ -0,0 +1,781 @@
+    def false_alarm_ratio(self, yes_category=2):
+        """Returns the false alarm ratio(s) for dichotomous contingency data
+
+            Parameters
+            ----------
+            yes_category : value, optional
+                The category coordinate value of the category corresponding to 'yes'
+
+            Returns
+            -------
+            xarray.Dataset or xarray.DataArray
+                An array containing the false alarm ratio(s)
+
+            Notes
+            -----
+            See http://www.cawcr.gov.au/projects/verification/
+        """
+
+        no_category = abs(yes_category - 2) + 1
+
+        hits = self.table.sel(
+            {
+                OBSERVATIONS_NAME + '_category': yes_category,
+                FORECASTS_NAME + '_category': yes_category,
+            },
+            drop=True,
+        )
+        false_alarms = self.table.sel(
+            {
+                OBSERVATIONS_NAME + '_category': no_category,
+                FORECASTS_NAME + '_category': yes_category,
+            },
+            drop=True,
+        )
+
+        return false_alarms / (hits + false_alarms)
+
+    @dichotomous_only
+    def false_alarm_rate(self, yes_category=2):
+        """Returns the false alarm rate(s) for dichotomous contingency data
+
+            Parameters
+            ----------
+            yes_category : value, optional
+                The category coordinate value of the category corresponding to 'yes'
+
+            Returns
+            -------
+            xarray.Dataset or xarray.DataArray
+                An array containing the false alarm rate(s)
+
+            Notes
+            -----
+            See http://www.cawcr.gov.au/projects/verification/
+        """
+
+        no_category = abs(yes_category - 2) + 1
+
+        false_alarms = self.table.sel(
+            {
+                OBSERVATIONS_NAME + '_category': no_category,
+                FORECASTS_NAME + '_category': yes_category,
+            },
+            drop=True,
+        )
+        correct_negs = self.table.sel(
+            {
+                OBSERVATIONS_NAME + '_category': no_category,
+                FORECASTS_NAME + '_category': no_category,
+            },
+            drop=True,
+        )
+
+        return false_alarms / (correct_negs + false_alarms)
+
+    @dichotomous_only
+    def success_ratio(self, yes_category=2):
+        """Returns the success ratio(s) for dichotomous contingency data
+
+            Parameters
+            ----------
+            yes_category : value, optional
+                The category coordinate value of the category corresponding to 'yes'
+
+            Returns
+            -------
+            xarray.Dataset or xarray.DataArray
+                An array containing the success ratio(s)
+
+            Notes
+            -----
+            See http://www.cawcr.gov.au/projects/verification/
+        """
+
+        no_category = abs(yes_category - 2) + 1
+
+        hits = self.table.sel(
```

Comment:
What about refactoring the code for hits, misses, etc. ? All scores basically use the same code there

Reply:
Nice suggestion. I've exposed `.hits`, `.misses`, `.false_alarms`, and `.correct_negatives` methods that are then used by the scoring methods. I've also added a direct test of these values into `test_dichotomous_scores()`